xmodule/split_test_block.py

```diff
@@ -172,10 +173,14 @@ def child_block(self):
     def child(self):
         """
         Return the user bound child block for the partition or None.
+
+        We are using try/except block here because we ran into issues with
+        CachingDescriptorSystem has no attribute when we get an icon for the split_test xblock.
         """
-        if self.child_block is not None:
+        try:
             return self.runtime.get_block_for_descriptor(self.child_block)
-        else:
+        except AttributeError:
+            log.warning("Error while getting block for descriptor")
```

Comment:
Please add a unique identifier (e.g. location), so it's possible to debug an error

Reply:
I also can't understand why you removed `if self.child_block is not None`
it means that you may break a case when `child_block` is None